### PR TITLE
PTX-14602 Fix Torpedo interfaces to bring them back to standards and rewrite upgrade volume driver and stork tests

### PR DESCRIPTION
**Pull request by nikolaypopov**
* Add new test for upgrading PX StorageCluster via upgrade hops using spec gen URL list
* Add new methods for upgrading and validating PX StorageCluster
* Rewriting a lot of volume driver func interfaces to the torpedo standards 
* Fix/Improve logging
* Fix/improve some returns
* Some go lint fixes  

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

**Comment by nikolaypopov** [comment]
Job is running here https://jenkins.pwx.dev.purestorage.com/job/Torpedo/view/Torpedo%20Upgrade/job/tp-upgrade-stc/2/console

**Review by CharudathGopal — APPROVED** [review]

**Comment by nikolaypopov** [comment]
Never mind I need to rewrite the logic and improve it a bit more, got some ideas from talking to @mborodin-px, will need some time

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Comment by nikolaypopov** [comment]
Had a successful run for both StorageCluster and Daemonset with Stork upgrade by default

File: drivers/volume/linstor/linstor.go [hunk 1]
```diff
@@ -3,17 +3,17 @@ package linstor
 import (
 	"context"
 	"fmt"
-	"github.com/portworx/torpedo/pkg/log"
 	"time"
 
+	"github.com/portworx/torpedo/pkg/log"
+
```
Comment (mborodin-px) [inline drivers/volume/linstor/linstor.go:9]: nit: remove this empty line

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -1343,8 +1275,7 @@ func (d *portworx) UpdateSharedv4FailoverStrategyUsingPxctl(volumeName string, s
 			TimeBeforeRetry: defaultRetryInterval,
 		})
 	if err != nil {
-		log.Errorf(fmt.Sprintf("error when creating local snapshot using PXCTL, Err: %v", err))
-		return err
+		return fmt.Errorf("failed to create local local snapshot using PXCTL, Err: %v", err)
```
Comment (mborodin-px) [inline drivers/volume/portworx/portworx.go:1278]: double `local` words

**Review by mborodin-px — CHANGES_REQUESTED** [review]
test Request changes

File: drivers/volume/portworx/portworx.go (hunk 2 above)
Comment (nikolaypopov) [reply drivers/volume/portworx/portworx.go]: Done

File: drivers/volume/linstor/linstor.go (hunk 1 above)
Comment (nikolaypopov) [reply drivers/volume/linstor/linstor.go]: Done

**Review by mborodin-px — APPROVED** [review]
LGTM

**Comment by nikolaypopov** [comment]
We have validated `operator` and `generic` upgrades here https://jenkins.pwx.dev.purestorage.com/job/Users/job/Maksym/job/tp-upgrade-pipeline04/105/console
I also ran multiple of my nextpx jobs with same image, no issues were found, will keep an eye on this after I merge

**Pull request merged**